utils/zoneinfo64/src/lib.rs

```diff
@@ -0,0 +1,689 @@
+            write!(f, "{:?}, ", (std as f64 / 3600.0, rule as f64 / 3600.0))?;
+            i += 1;
+        }
+        for &(hi, lo) in self.trans_post32 {
+            dbg_timestamp(f, ((hi as u32 as u64) << 32 | (lo as u32 as u64)) as i64)?;
+            let (std, rule) = self.type_offsets[self.type_map[i] as usize];
+            write!(f, "{:?}, ", (std as f64 / 3600.0, rule as f64 / 3600.0))?;
+            i += 1;
+        }
+        write!(f, "], ")?;
+
+        write!(f, "}}")
+    }
+}
+
+impl<'a> ZoneInfo64<'a> {
+    pub fn try_from_u32s(resb: &'a [u32]) -> Result<Self, BinaryDeserializerError> {
+        use serde::de::*;
+
+        #[derive(Debug, Deserialize)]
+        #[serde(rename = "zoneinfo64")]
+        #[serde(rename_all = "PascalCase")]
+        struct ZoneInfo64Raw<'a> {
+            #[serde(borrow)]
+            zones: Vec<TzZoneRaw<'a>>,
+            #[serde(borrow, deserialize_with = "resb::binary::helpers::vec_utf_16")]
+            names: Vec<&'a PotentialUtf16>,
+            #[serde(borrow, deserialize_with = "rules")]
+            rules: Vec<(&'a str, TzRule)>,
+            #[serde(deserialize_with = "regions")]
+            regions: Vec<Region>,
+        }
+
+        #[derive(Debug)]
+        pub enum TzZoneRaw<'a> {
+            Table(TzZoneDataRaw<'a>),
+            Int(u32),
+        }
+
+        impl<'de: 'a, 'a> Deserialize<'de> for TzZoneRaw<'a> {
+            fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+            where
+                D: serde::Deserializer<'de>,
+            {
+                struct TzDataRuleEnumVisitor<'a> {
+                    phantom: PhantomData<TzZoneRaw<'a>>,
+                }
+
+                impl<'de: 'a, 'a> Visitor<'de> for TzDataRuleEnumVisitor<'a> {
+                    type Value = TzZoneRaw<'a>;
+
+                    fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+                        formatter.write_str("an unsigned 32-bit integer or a table of rule data")
+                    }
+
+                    fn visit_u32<E>(self, v: u32) -> Result<Self::Value, E>
+                    where
+                        E: Error,
+                    {
+                        Ok(TzZoneRaw::Int(v))
+                    }
+
+                    fn visit_map<A>(self, map: A) -> Result<Self::Value, A::Error>
+                    where
+                        A: MapAccess<'de>,
+                    {
+                        let value =
+                            TzZoneDataRaw::deserialize(value::MapAccessDeserializer::new(map))?;
+
+                        Ok(TzZoneRaw::Table(value))
+                    }
+                }
+
+                deserializer.deserialize_any(TzDataRuleEnumVisitor {
+                    phantom: PhantomData,
+                })
+            }
+        }
+
+        #[derive(Debug, Deserialize)]
+        #[serde(rename_all = "camelCase")]
+        pub struct TzZoneDataRaw<'a> {
```

Comment:
nit: this is a _lot_ of stuff living inside a function, I think we should just have a `serde` or `parse` module that has all of this